### PR TITLE
Support timezones as UTC offset in minutes

### PR DESCRIPTION
**Pull request by cibeah**
Part 3 of craft-ai/craft-ai-client-js#130
Support timezones as UTC offset in minutes in the python client

### CODE REVIEW

File: craftai/timezones.py [hunk 1]
```diff
@@ -36,11 +37,21 @@
 }
 
 def is_timezone(value):
+  if isinstance(value, six.integer_types) and value <= 840 and value >= -720:
+    return True
+  if not isinstance(value, six.string_types):
+    return False
   result_reg_exp = _TIMEZONE_REGEX.match(value) is not None
```
Comment (alexisfontaine) [inline craftai/timezones.py]: You could prevent systematic matching of the regular expression here by lazy boolean evaluation.

File: .pylintrc [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [MASTER]
 
-disable=missing-docstring,locally-enabled,locally-disabled,duplicate-code
+disable=missing-docstring,locally-enabled,locally-disabled,duplicate-code, unnecessary-lambda
```
Comment (alexisfontaine) [inline .pylintrc]: Is that working with a space here?

Comment (cibeah) [reply .pylintrc]: Yes, I updated it

File: craftai/timezones.py [hunk 3]
```diff
@@ -53,3 +73,13 @@ def timezone_offset_in_sec(timezone):
     offset = -offset
 
   return offset
+
+def timezone_offset_in_standard_format(timezone):
+  if isinstance(timezone, six.integer_types):
+    sign = '+' if timezone >= 0 else '-'
```
Comment (Totoketchup) [inline craftai/timezones.py]:
```suggestion
    sign = "+" if timezone >= 0 else "-"
```

File: craftai/timezones.py [hunk 4]
```diff
@@ -53,3 +73,13 @@ def timezone_offset_in_sec(timezone):
...
+    absolute_offset = abs(timezone)
+    if absolute_offset < 16:
+      return  '%s%02d:00' % (sign, absolute_offset)
```
Comment (Totoketchup) [inline craftai/timezones.py]:
```suggestion
      return  "%s%02d:00" % (sign, absolute_offset)
```

File: craftai/timezones.py [hunk 5]
```diff
@@ -53,3 +73,13 @@ def timezone_offset_in_sec(timezone):
...
+    return '%s%02d:%02d' % (sign, int(absolute_offset / 60),
```
Comment (Totoketchup) [inline craftai/timezones.py]:
```suggestion
    return "%s%02d:%02d" % (sign, int(absolute_offset / 60),
```

File: craftai/timezones.py [hunk 6]
```diff
@@ -36,11 +37,30 @@
...
+  if not isinstance(value, six.string_types):
+    return False
+  if value in TIMEZONES:
+    return True
   result_reg_exp = _TIMEZONE_REGEX.match(value) is not None
-  result_abbreviations = value in TIMEZONES
-  return result_reg_exp or result_abbreviations
+  return result_reg_exp
+
+# def get_timezone_key(configuration):
```
Comment (Totoketchup) [inline craftai/timezones.py]: delete ?

File: craftai/timezones.py [hunk 7]
```diff
@@ -36,11 +37,30 @@
...
+
+def get_timezone_key(configuration):
+  for key in configuration:
+    if configuration[key]["type"] == "timezone":
+      return key
+  return None
 
 def timezone_offset_in_sec(timezone):
+  if isinstance(timezone, six.integer_types):
+    # If the UTC offset is between -16 and +16, consider it as a
```
Comment (Totoketchup) [inline craftai/timezones.py]: -15 / 15 or -16 / 16
It the sampling you mention '-15 15' with including signs.
It's exactly the same of course to pu < 16 and <= 15, but it's just to be fitting with learning_sampling here.

Comment (cibeah) [reply craftai/timezones.py]: You're right, did just that

**Pull request merged**